File: backend/payment/src/main/java/com/avocado/payment/repository/ConsumerRepository.java
package com.avocado.payment.repository;

import com.avocado.payment.entity.Consumer;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.UUID;

@Repository
public interface ConsumerRepository extends JpaRepository<Consumer, UUID> {
}

File: backend/user-server/src/main/kotlin/com/avocado/userserver/db/repository/ConsumerRepository.kt
package com.avocado.userserver.db.repository

import com.avocado.userserver.api.service.SocialType
import com.avocado.userserver.db.entity.Consumer
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface ConsumerRepository : CoroutineCrudRepository<Consumer, ByteArray> {

    suspend fun findBySubAndSocial(sub:String, social:SocialType): Consumer?

    suspend fun save(consumer: Consumer): Consumer

}